README.adoc

```diff
@@ -60,6 +61,55 @@ Then, run:
 
 This will generate the `result.sarif` file containing the query results.
 
+==== Self-contained script
+
+The following shell script creates a database by running the specified build command, analyzes the database with the specified queries only, and then uploads the results, excluding any suppressed findings, to GitHub.
+
+[source,bash]
+----
+#!/usr/bin/env bash
+set -e errexit
+set -e nounset
+set -e pipefail
+[[ -f pom.xml ]] || { echo "This script must be run from a Maven project directory" ; exit 1 ; }
+[[ -v "GITHUB_TOKEN" ]] || { echo "GITHUB_TOKEN is undefined. " ; exit 1 ; }  # <1>
+[[ -v "GH_REPO" ]] || { echo "GH_REPO is undefined" ; exit 1 ; }              # <2>
+[[ -v "GH_REF" ]] || { echo "GH_REF is undefined." ; exit 1 ; }               # <3>
+[[ -v "GH_SHA" ]] || { echo "GH_SHA is undefined." ; exit 1 ; }               # <4>
+for TOOL in codeql jq mvn ; do
+    which "$TOOL" >/dev/null || { echo "$TOOL not found on PATH" ; exit 1 ; }
+done
+TEMPDIR="$( mktemp -d -t jenkins-codeql.XXXX )"
+codeql database create "$TEMPDIR"/codeql-java-database \
+  --language=java \
+  --command='mvn clean verify -Pquick-build'                                  # <5>
+codeql database analyze "$TEMPDIR"/codeql-java-database \
+  --sarif-add-query-help \
+  --format=sarif-latest \
+  --output="$TEMPDIR"/result.sarif \
+  --download \
+  jenkins-infra/jenkins-codeql \                                                <6>
```

Comment:
The comment on this line is a little awkward because it prevents use after copying & pasting without any modifications.